src/lib.rs

```diff
@@ -241,33 +241,30 @@ impl<'a> fmt::Debug for Segment<'a> {
     }
 }
 
-impl<'a> ObjectSegment<'a> for Segment<'a> {
-    fn address(&self) -> u64 {
+impl<'a> Segment<'a> {
+    fn inner(&self) -> &ObjectSegment<'a> {
```

Comment:
`#[inline]`?

Not worried about introducing virtual calls?

Reply:
#[inline] won't help avoid the virtual calls: https://github.com/rust-lang/rust/issues/45774

Reply:
I assumed that rust would inline this automatically since it's not public or a trait method?

I hadn't considered the optimization of virtual calls though. I'll address that in a followup PR. Thanks for investigating that @jrmuizel.